### ISSUE TITLE
[Bug Report] Logit attribution produces different values for Gemma2

### ISSUE THREAD

**Issue by wj210** [TransformerLensOrg/TransformerLens#828]
If you are submitting a bug report, please fill in the following details and use the tag [bug].

**Describe the bug**
When doing logit attribution for Gemma2 models, the unembedding of the final residual output doesn't tally with the logits in the original forward pass.

**Code example**
```
model = HookedTransformer.from_pretrained(
   "google/gemma-2-2b-it",
    center_unembed=False,
    center_writing_weights=False,
    fold_ln=True,
    refactor_factored_attn_matrices=False,
    default_padding_side = 'left',
    torch_dtype = t.bfloat16,
    default_prepend_bos = False if 'chat' in model_name else True,
)

prompt= "The Eiffel Tower is in the city of"
answer = 'Paris'
other_answer = 'Rome'

prompt = model.tokenizer.apply_chat_template([{'role':'user','content':prompt}],add_generation_prompt=True,tokenize=False)

# prompt_token = model.to_tokens(prompt)
prompt_token = model.tokenizer(prompt,return_tensors='pt')['input_ids'].to(device)
ans_token = model.to_tokens(answer).item()
other_ans_token = model.to_tokens(other_answer)[0].item()

ans_resi_direction = model.tokens_to_residual_directions(t.tensor([ans_token,other_ans_token])).float()

logit_diff_resi_direction = ans_resi_direction[0] - ans_resi_direction[1]

with t.no_grad():
    out,cache = model.run_with_cache(prompt_token)

ori_logit = out[0,-1,ans_token]
other_logit = out[0,-1,other_ans_token]
original_logit_diff = ori_logit - other_logit

final_stream = cache["resid_post", -1][:,-1,:]
scaled_stream = cache.apply_ln_to_stack(final_stream, layer=-1, pos_slice=-1).float()[0]

ll_logit = einops.einsum(
    scaled_stream, logit_diff_resi_direction,
    "d_model,  d_model ->")

print ('Original logit diff: ',original_logit_diff)
print ('Unembedded residual stream ',ll_logit)

```

This produces 8.3 and 9 for the two logits, which has quite a significant issue. This doesn't happen with Llama models.  Running in full precision has the same issue as well.

**System Info**
torch==2.4.1
transformers=4.45

**Comment by wj210** [comment]
nvm. forgot that for gemma, we need to softcap them. 
just apply the below function to the unembedded logit
```
def softcap_logits(logits):
    if model.cfg.output_logits_soft_cap > 0:
        logits = logits / model.cfg.output_logits_soft_cap
        logits = t.tanh(logits)
        logits = logits * model.cfg.output_logits_soft_cap
    return logits
```

**Comment by qwenzo** [comment]
I don't think you need to do this manually.  run_with_cache should do that for you, no?
https://github.com/TransformerLensOrg/TransformerLens/blob/d7f9eb1655b810c133ac83b39a2b19a6cac6945c/transformer_lens/HookedTransformer.py#L631

**Comment by wj210** [comment]
not if u are doing manual logit lens projection. it doesnt go through the forward function.
